### PR TITLE
EOS-20659: Field Prepare: Time configuration using cortx_setup cli command

### PR DESCRIPTION
**Pull request by rahuljyoti65**
Signed-off-by: Rahul Jyoti <rahul.jyoti@seagate.com>

### CODE REVIEW

File: lr-cli/cortx_setup/commands/node/prepare/time.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        """Time configuration command execution method.
+
+        Execution:
+        `cortx_setup node prepare time --server <ntp-server> --timezone <timezone>`
+        """
+        ntp_server = kwargs.get('server')
+        ntp_timezone = kwargs.get('timezone')
+        self.logger.info(f"Configuring Time")
+        try:
+            set_ntp(server=ntp_server, timezone=ntp_timezone)
```
Comment (mazinamdar) [inline lr-cli/cortx_setup/commands/node/prepare/time.py:47]: set_ntp will set time server for system and controller so can we have logic of updating date of only system/node in this API ?

@ypise  can you please suggest , 
1. We can update pillar for server and timezone in system.sls
2. from this API call states which update time for system/node only.

Comment (rahuljyoti65) [reply lr-cli/cortx_setup/commands/node/prepare/time.py]: comment addressed as discussed

**Review by mazinamdar — APPROVED** [review]
looks good to me

File: lr-cli/cortx_setup/commands/node/prepare/time.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+        PillarSet().run(
+            'system/ntp/time_zone',
+            ntp_timezone,
+            targets=node_id,
+            local=True
+        )
+
+        try:
+            self.logger.info(f"Setting time on node with server={ntp_server} & timezone={ntp_timezone}")
```
Comment (ypise) [inline lr-cli/cortx_setup/commands/node/prepare/time.py:86]: Convert this to debug log.
Ideally we would want this to be equivalent to verbose.

File: lr-cli/cortx_setup/commands/node/prepare/time.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            self.set_server_time()
+            chassis = grains_get("hostname_status:Chassis")[node_id]["hostname_status:Chassis"]
+            if chassis == "server":
```
Comment (ypise) [inline lr-cli/cortx_setup/commands/node/prepare/time.py:89]: We have decided to use the enclosure id instead, haven't we?
```
if not machine_id in enclosure_id:
    self.logger.debug("")
    self.set_enclosure_time()
```

File: lr-cli/cortx_setup/commands/node/prepare/time.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                self.logger.info(f"Setting time on enclosure with server={ntp_server} & timezone={ntp_timezone}")
```
Comment (ypise) [inline lr-cli/cortx_setup/commands/node/prepare/time.py:90]: All logs in this file should be debug or error.

File: lr-cli/cortx_setup/commands/node/prepare/time.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                self.set_enclosure_time()
+        except Exception as e:
+            print(e)
+            self.logger.error("Time configuration failed")
```
Comment (ypise) [inline lr-cli/cortx_setup/commands/node/prepare/time.py:94]: print(e) not necessary.
use `self.logger.exception` instead.

**Review by ypise — CHANGES_REQUESTED** [review]
@rahuljyoti65 
Address the suggested comments.
Also, the check for enclosure time set should be purely based on valid enclosure ID

File: lr-cli/cortx_setup/commands/node/prepare/time.py (hunk 3 above)
Comment (mazinamdar) [reply lr-cli/cortx_setup/commands/node/prepare/time.py:89]: @ypise Actually we thought our main target is to just check it is HW or vm, so I asked him to use that.
If there are other cases then we can change to above code block.

File: lr-cli/cortx_setup/commands/node/prepare/time.py (hunk 4 above)
Comment (rahuljyoti65) [reply lr-cli/cortx_setup/commands/node/prepare/time.py]: Done

File: lr-cli/cortx_setup/commands/node/prepare/time.py (hunk 3 above)
Comment (rahuljyoti65) [reply lr-cli/cortx_setup/commands/node/prepare/time.py]: Done

File: lr-cli/cortx_setup/commands/node/prepare/time.py (hunk 2 above)
Comment (rahuljyoti65) [reply lr-cli/cortx_setup/commands/node/prepare/time.py]: Done

**Review by ypise — APPROVED** [review]

**Pull request merged**